src/segment.rs

```diff
@@ -270,6 +315,34 @@ impl<'a> Drop for Segment<'a> {
     }
 }
 
+#[cfg(feature = "distributed_tracing")]
+struct FreeableString(*mut std::os::raw::c_char);
+
+#[cfg(feature = "distributed_tracing")]
+impl FreeableString {
+    fn new(inner: *mut std::os::raw::c_char) -> Self {
+        Self(inner)
+    }
+
+    fn convert(&self) -> Result<String> {
+        let c_str = unsafe { std::ffi::CStr::from_ptr(self.0) };
+
+        Ok(c_str
+            .to_str()
+            .expect("This should really never happen...")
```

Comment:
This should either utilize a new `Utf8Error` variant of the `newrelic::Error` type and return something like

```
        c_str
            .to_str()
            .map(ToString::to_string)
            .map_err(Error::Utf8Error)
```

...or should just continue using `.expect()` and return a `String` (instead of `Result<String>`), since we're getting a HTTP safe payload anyway.